### ISSUE TITLE
Using format breaks the cursor position / keyboard input

### ISSUE THREAD

**Issue by kasbah** [vlad-ignatov/react-numeric-input#14]
If you add a format function that returns a string it breaks the user being able to type in numbers. It only allows you to type in one digit and the value can disappear. The cursor seems to move to the end.

You can see the in the Buy Parts section of [this page](http://quantity-input.preview.kitnic.it/boards/github.com/JarrettR/USBvil/) for instance.

Here is a video: https://kaspar.h1x.com/numeric-input-bug1.webm

Edit:

Sorry forgot to include example code. The above is rendered using:

```jsx
<NumericInput value={this.state.buyMultiplier.times} />
<NumericInput value={this.state.buyMultiplier.percent} format={(n) => `${n}%`} />
```

Full source [here](https://github.com/monostable/kitnic/blob/quantity-input/src/buy_parts.jsx#L185)

**Comment by kasbah** [comment]
Looks like this is related to https://github.com/facebook/react/issues/955

**Comment by puneet-sutar** [comment]
@kasbah I faced the same issue. I went through the implementation and it seems you have to provide a parsing function(to returned the unformatted number). This fixed the issue for me.

 Sample Code:

```javascript
<NumericInput  parse={removeNonNumericChar} format={numberWithCommas} />
```
```javascript
function removeNonNumericChar(x){
  return x.toString().replace(/[^\/\d]/g,'')
}
function numberWithCommas(x){
  return x ? x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",") : x;
}
```

Hope this help 😄

**Comment by vlad-ignatov** [comment]
@puneet-sutar you are correct that `parse` and `format` should go in pairs but that is only when `format` produces something that the component cannot parse. The widget uses `parseFloat` internally which should handle things like `50%` without having to use additional `parse` function.

This issue was related to the selection not being preserved and should be fixed in 2.0.5. You can try it [here](http://vlad-ignatov.github.io/react-numeric-input/examples/v2.0.5/index.html) in the `Custom format` section.
